### ISSUE TITLE
Issue with PubSub client using provided JWTAccess 

### ISSUE THREAD

**Issue by salrashid123** [googleapis/google-auth-library-nodejs#1524]

**Comment by feywind** [comment]
Comments from @alexander-fenster :
* looks like the JWT AuthClient does not provide a proper getHeaders() for gRPC to use
* one thing to check is to try with fallback: "rest" to remove gRPC from the picture - the auth logic will not change and should still work (or fail in a similar way)

**Comment by alexander-fenster** [comment]
Actually, after I looked more into this, I'm assuming that the `PubSub` constructor, which takes all parameters mentioned in [`GrpcClientOptions`](https://github.com/googleapis/gax-nodejs/blob/c6286cb73b6cdb5afa4cdb5df51a3deb222b0113/src/grpc.ts#L47-L53), actually expects to see an `auth` parameter that is set to an instance of [`GoogleAuth`](https://github.com/googleapis/google-auth-library-nodejs/blob/main/src/auth/googleauth.ts#L130) class.

Long story short, what I would try is:  _(typing right here, please forgive me for possible syntax errors)_

```ts
import {GoogleAuth} from 'google-auth-library';  // or from 'google-gax'

const auth = new GoogleAuth({authClient});
const pubsub = new PubSub({auth});
```

This should make it call `getRequestHeaders` properly. If not, please ping me and we'll figure out.

**Comment by feywind** [comment]
@alexander-fenster I think this is actually possible:

```
  authClient: jwtClient,
```

The types are really confusing, but there are some `extends` in there that do allow for it. I guess if that's not the right way, maybe we should look at fixing those types?

**Comment by danielbankhead** [comment]
I think @alexander-fenster is right; the `GrpcClientOptions` expects an instance of `GoogleAuth` rather than `AuthClient`. Technically, the auth library is behaving as intended (nothing to fix/change here).
